var/spack/repos/builtin/packages/mivisionx/package.py

```diff
@@ -31,25 +32,38 @@ def url_for_version(self, version):
 
     variant('build_type', default='Release', values=("Release", "Debug"), description='CMake build type')
 
+    resource(name='opencv',
+             url='https://github.com/opencv/opencv/archive/3.4.6.tar.gz',
+             sha256='e7d311ff97f376b8ee85112e2b536dbf4bdf1233673500175ed7cf21a0089f6d',
+             expand=True,
+             destination='',
+             placement='opencv',
+             when='@4.0.0:')
+
     def flag_handler(self, name, flags):
         spec = self.spec
         protobuf = spec['protobuf'].prefix.include
+        opencv_incl = join_path(self.stage.source_path,
+                                'opencv/modules/highgui/include')
         if name == 'cxxflags':
             flags.append('-I{0}'.format(protobuf))
+            flags.append('-I{0}'.format(opencv_incl))
         return (flags, None, None)
 
     depends_on('ffmpeg@4.1.1', type='build')
     depends_on('protobuf@3.5.0', type='build')
-    depends_on('opencv@3.4.6 +calib3d+core+features2d+highgui+imgproc+video+videoio+videostab', type='build')
+    depends_on('opencv@3.4.6 +calib3d+core+features2d+highgui+imgcodecs+imgproc+video+videoio+videostab+flann', type='build')
```

Comment:
So you have both a dependency on opencv as well as a resource? Why do you need both?

Reply:
headers file(eg:-highgui_c.h) part of source are not part of install include directory. hence i have added that as a resource. Do you see any better way of doing this?

Reply:
Hmm, @bvanessen is also helping me maintain `opencv`, maybe he has an idea. I'm surprised this library decided to use internal headers that aren't normally installed.

Reply:
I believe this was available previously in the install path ....before revamp of opencv... if my memory serves me right....
@bvanessen Will you be able comment here ?.

Reply:
If that's the case then we could force the build to use an older version of OpenCV until the developers make it work for newer OpenCV.

Reply:
Sorry i meant... older version of recipe...rather...We were using the same version of opencv(3.4.6) prior as well.....i know there are some changes which happened on this recipe.. recently. I am not completely sure if this could be the problem...

Reply:
There have been a lot of recent revamps to the `opencv` package, but none that change what gets installed. The default variants did change. But if you explicitly depend on `opencv+highgui` it should still install that part. In my build of OpenCV 4.5.0 I see: `include/opencv4/opencv2/highgui/highgui_c.h`. Is this the file you're looking for?

Reply:
Seems after branch update.... i am able to  get that file - highgui_c.h, so i have removed the resource now. I have tried it previously multiple times...I donno what was the issue. 

Please go ahead provide your review and approval.